backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EventTypeRepository.java

```diff
@@ -0,0 +1,40 @@
+package com.redhat.cloud.notifications.db.repositories;
+
+import com.redhat.cloud.notifications.models.EventType;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.ws.rs.NotFoundException;
+
+@ApplicationScoped
+public class EventTypeRepository {
+
+    @Inject
+    EntityManager entityManager;
+
+    /**
+     * Find an event type by its exact name.
+     * @param eventTypeName the name of the event type to look by.
+     * @return the found event type.
+     */
+    public EventType findByName(final String eventTypeName) {
+        final String findByName =
+            "SELECT " +
+                "e " +
+            "FROM " +
+                "EventType AS e " +
+            "WHERE " +
+                "e.name = :name";
+
+        try {
+            return this.entityManager.createQuery(findByName, EventType.class)
+                    .setParameter("name", eventTypeName)
+                    .getSingleResult();
+        } catch (final NoResultException e) {
+            throw new NotFoundException("the specified event type name was not found");
```

Comment:
Could you please return an `Optional.empty()` here and throw the exception at a higher level in the stack?

Reply:
This is no longer relevant, since the method is not required.